src/main/java/com/redhat/cloud/custompolicies/app/model/engine/FullTrigger.java

```diff
@@ -90,25 +95,52 @@ private static String generateId() {
     return UUID.randomUUID().toString();
   }
 
+  public void updateFromPolicy(Policy policy) {
+    if (policy.actions == null || policy.actions.isEmpty()) {
+      trigger.actions.clear();
+    }
+    String[] actionsIn = policy.actions.split(";");
+    Set<TriggerAction> newActions = new HashSet<>();
+    for (String actionIn : actionsIn) {
+      if (actionIn.trim().isEmpty()) {
+        continue;
+      }
+      String actionName;
+      if (actionIn.contains(" ")) {
+        actionName = actionIn.substring(0, actionIn.indexOf(' '));
+      }
+      else {
+        actionName = actionIn;
+      }
+      actionName = actionToTriggerActionMap.get(actionName);
+
+      boolean found = false;
+      for (TriggerAction ta : trigger.actions) {
+        String taName = ta.actionPlugin;
+        if (taName.equals(actionName)) {
+          newActions.add(ta);
+          found = true;
+        }
+      }
+      if (!found) {
+        var ta = new TriggerAction();
+        ta.actionPlugin = actionName;
+        newActions.add(ta);
+      }
+    }
+    trigger.actions.removeIf(ta -> !newActions.contains(ta));
```

Comment:
Should we remove this line?
As I understand, `newActions` contains the actions that we want to keep and the new actions.

Reply:
Yes. So we need to remove from `trigger.actions` those that are not longer wanted, which that line does. As alternative, one could do `trigger.actions = newActions;`